src/main/java/app/model/agents/WallFollow/WfWorld.java

```diff
@@ -75,4 +75,23 @@ public void markWallAsCovered(GraphCell cell, Vector agentPos)
             getVerticalWallsCovered().add(cell.getY());
         }
     }
+
+    @Override
+    public GraphCell getNonObstacleNeighbour(GraphCell vertex)
+    {
+        // TODO "Cannot invoke "app.model.agents.Cells.GraphCell.getObstacle()" because "vertex" is null"
+        //  probably the issue of being on the border of the map
+        if (vertex.getObstacle())
```

Comment:
wouldn't replacing it by: if(vertex != null && vertex.getObstacle()) solve the NullPointerError?

(if it sees the first one is false in AND-statement, it automatically doesn't check the second condition)